### ISSUE TITLE
[QUESTION]: Correct way of obtaining graph embeddings

### ISSUE THREAD

**Issue by amirnikooie** [chemprop/chemprop#409]
Hello,
I did some quick search in previous issues as well as the documentation, but couldn't find a proper response to my question. I have a pre-trained chemprop model for both MPNN and FFN part, but what I need is graph embeddings for any given data. In other words, the output of MPNN right before the FFN. I have a way of running it but wasn't sure it is a correct way. Could you please confirm if this is the way to obtain those embeddings? I have put the code snippet below. 

Some explanation about the code: The `input_file` below is the file that contains SMILES strings of molecules I need the embeddings for and a dummy response vector as the second column. The `ge_out.npy` is the matrix containing those graph embeddings saved as a numpy file, whose size is already determined during the training stage, and is 1200. Since I had additional features, Morgan, during training, I added them here as well.

```
       import numpy as np
       import chemprop

	arguments = [
	'--test_path', input_file, 
	'--preds_path', 'temp_out.csv',
	'--checkpoint_dir', './chemprop/train_ouputs',
	'--features_generator', 'morgan',
	'--fingerprint_type', 'MPN'
	]
	args = chemprop.args.FingerprintArgs().parse_args(arguments)
	embeddings = chemprop.train.molecule_fingerprint.molecule_fingerprint(args)
        np.save('ge_out.npy', embeddings)
```

Any feedback on the way those embeddings are obtained or a better way of doing this is appreciated.
Thanks!

**Comment by kevingreenman** [comment]
Thanks for your question. Your approach looks correct, except that providing `'--features_generator', 'morgan'` is not necessary since you're using the `MPN` fingerprint. Providing features is only necessary if you're using the `last_FFN` `fingerprint_type` because these features are concatenated with the MPN representation before being fed to the FFN network. Thus, the molecule features affect the FFN output but do not affect the MPN output. 

There's an example of fingerprinting in Python in our [example notebook](https://github.com/chemprop/chemprop/blob/master/colab_demo.ipynb), and it looks like your code matches that.

**Comment by amirnikooie** [comment]
Thank you, @kevingreenman , for your prompt reply and thorough explanation. This helps!
One quick question though. It seems the documentation page in this [link](https://chemprop.readthedocs.io/en/latest/index.html) does not contain all the information or at least some useful functions you discuss in README if GitHub or example notebook you mentioned above are not explained there at all. Transfer learning, obtaining scripts, etc. are some examples. So does this mean that webpage is not up to date or should be ignored for now? Because that is the first place some would go before checking the GitHub, and it may be good to link some of those up to date materials in that web page too.
Thanks again for all you and the team do for this package!

**Comment by kevingreenman** [comment]
That's correct, the readthedocs is not currently up-to-date. Hopefully we'll be able to update that eventually, but for now I think the README and example notebook should be the first places you look.
